ports/stm32/boards/LEGO_HUB_NO7/README.md

```diff
@@ -0,0 +1,135 @@
+cannot be erased.  This bootloader is entered by holding down the button for 5 seconds,
+at which point the USB DFU device appears.  If the battery is installed then the
+RGB LED will flash purple. If the battery is not installed, the LED will flash orange
+briefly and then the hub will turn off (so having the battery installed is required).
+When this bootloader is active, the flash from 0x08008000 and up can be erased
+and programmed via USB DFU.
+
+The built-in bootloader has some drawbacks: it cannot be entered programmatically,
+and it does not keep the Hub powered up when running from battery (which requires
+keeping BAT_PWR_EN high).  As such, this board is configured to work with mboot as
+a secondary bootloader: mboot is placed at 0x08008000 and the main application
+firmware at 0x08010000.  When mboot is installed it can be entered programatically
+via machine.bootloader().
+
+Backing up original Hub firmware
+--------------------------------
+
+Before installing MicroPython it is advised to backup the original LEGO firmware that
+the Hub comes installed with.  To do this, enter the built-in bootloader by holding
+down the Bluetooth button for 5 seconds while powering up the Hub via USB (you may
+need to take out the battery and disconnect USB to power off the Hub first).  Then
+run the following command from the root of this repository:
+
+    $ cd ports/stm32
+    $ make BOARD=LEGO_HUB_NO7 backup-hub-firmware
+
+This will create a file called `lego_hub_firmware.dfu`.  Put this file in a safe
+location.  To restore it, enter the built-in bootloader again and run:
+
+    $ make BOARD=LEGO_HUB_NO7 restore-hub-firmware
+
+This will restore the original firmware but not the filesystem.  To recreate the
+original filesystem the Hub must be updated using the appropriate LEGO PC
+application.
+
+Installing MicroPython
+----------------------
+
+You first need to build and install mboot, which only needs to be done once.  From
+the root of this repository run:
+
+    $ cd ports/stm32/mboot
+    $ make BOARD=LEGO_HUB_NO7
+
+Now enter the built-in bootloader by holding down the Bluetooth button for 5
+seconds while powering up the Hub via USB (you may need to take out the battery
+and disconnect USB to power off the Hub first).  Then run:
+
+    $ make BOARD=LEGO_HUB_NO7 deploy
+
+mboot should now be installed.  To enter mboot, remove USB and the battery.
+Connect the USB cable (the other end of the USB cable must be connected to
+something that provides power). The status light should start cycling through
+different colors. Replace the battery (the button will not work without the
+battery present). Press the button to activate the desired boot mode:
+
+- Status light is red - run application (normal boot).
+- Status light is green - run application in factory file system mode.
+- Status light is blue - run application in safe mode.
+- Status light is white - start DFU on the USB port.
```

Comment:
Are there standard colors for the bootloader modes? I found something that says white is DFU mode on one of the Pyboards, but that was all.

Reply:
Yes, white is used on PYBD.